client/my-sites/checkout/composite-checkout/components/contact-details-container.tsx

```diff
@@ -53,7 +54,12 @@ export default function ContactDetailsContainer( {
 		.filter( ( product ) => ! isDomainMapping( product ) )
 		.map( getDomain );
 	const checkoutActions = useDispatch( 'wpcom-checkout' );
```

Comment:
Does this need explicit types also? It might not but I wanted to check. There's already a check down below (and everywhere else we call `useDispatch` for this store) in case `checkoutActions` is falsy but its type appears to be `DispatcherMap` 🤷 

(Same question for all other instances of `useDispatch( 'wpcom-checkout' )`)

Reply:
You're right, it does look like we're loosing types for `useDispatch` now, maybe across the repo. 

The core problem is similar to before, where we need to update to this new version of `@wordpress/data` in order to get the built-in type enhancements. But by upgrading, we have to ditch our manual type overrides, so we loose some type info and encounter a ton of type errors. I think the changes in this PR are mostly oriented to just fixing all of the issues `tsc` reports so that we can move forward.

The real solution is to migrate everything to the new approach with `register`, which is definitely where we want to end up. But it's challenging to do that here due to the scope and size of the update.

So our mindset is that this PR puts us in an intermediate place where we still have work to do, but ultimately makes it somewhat easier to get to the finished state

Reply:
And I guess the side problem is that `@wordpress/data` doesn't offer a very straightforward upgrade path, which leaves us with limited options, none of which are that good :/

Reply:
Yeah, I'm not too worried about this one since it's already missing. I just wanted to call it out. Feel free to leave this alone for now.

Reply:
What @noahtallen said. Part of the problem is that there are too many things to be changed to adhere to the new types and it has to be done incrementally. We're happy to aid in further improving those types and we'll most likely have to as we keep upgrading to the latest `@wordpress` packages and more of those start providing their own types. We very much see this PR as a first step.